src/lib/components/Button.svelte

```diff
@@ -0,0 +1,46 @@
+<script lang="ts">
+	export let type: 'default' | 'rounded' | 'square' = 'default';
+</script>
+
+<button class="btn {type}" on:click>
```

Comment:
You can rather write `class='btn' class:type`.

Also, I find it easier to have the base class for a component named exactly as the component's name.
That would lead to `class='Button' class:type`

Reply:
`class:type` would add a class type if the "type" variable isn't empty but not the content of the variable.